src/devices/Shtc3/Shtc3.cs

```diff
@@ -0,0 +1,282 @@
+                }
+
+                switch (value)
+                {
+                    case Status.Idle:
+                        // Change to idle awake sensor
+                        Wakeup();
+                        break;
+                    case Status.Sleep:
+                        // Change to sleep making sensor in sleeping mode
+                        Sleep();
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+
+                _status = value;
+            }
+        }
+
+        /// <summary>
+        /// Try read Temperature and Humidity
+        /// </summary>
+        /// <param name="temperature">Temperature returned by sensor</param>
+        /// <param name="relativeHumidity">Humidity return by sensor</param>
+        /// <param name="lowPower">"true" measured in low power mode, "false"(default) measured in normal power mode</param>
+        /// <param name="clockStretching">"true" allow clock stretching, "false" (default) without clock stretching</param>
+        /// <returns>True if operation was successful</returns>
+        public bool TryGetTemperatureAndHumidity(out Temperature temperature, out Ratio relativeHumidity, bool lowPower = false, bool clockStretching = false)
+        {
+            Register cmd = GetMeasurementCmd(lowPower, clockStretching);
+
+            if (!TryReadSensorData(cmd, out var st, out var srh))
+            {
+                temperature = default(Temperature);
+                relativeHumidity = default(Ratio);
+                return false;
+            }
+
+            // Details in the Datasheet P9
+            temperature = Temperature.FromDegreesCelsius(Math.Round(st * 175 / 65536.0 - 45, 1));
+            relativeHumidity = Ratio.FromDecimalFractions(srh / 65536.0);
+            return true;
+        }
+
+        private bool TryReadSensorData(Register cmd, out double temperature, out double humidity)
+        {
+            Write(cmd);
+
+            Span<byte> readBuff = stackalloc byte[6];
+
+            _i2cDevice.Read(readBuff);
+
+            // Details in the Datasheet P7
+            int st = BinaryPrimitives.ReadInt16BigEndian(readBuff.Slice(0, 2));      // Temp
+            int srh = BinaryPrimitives.ReadUInt16BigEndian(readBuff.Slice(3, 2));     // Humi
+
+            // check 8-bit crc
+            bool tCrc = CheckCrc8(readBuff.Slice(0, 2), readBuff[2]);
+            bool rhCrc = CheckCrc8(readBuff.Slice(3, 2), readBuff[5]);
+            if (!tCrc || !rhCrc)
+            {
+                temperature = double.NaN;
+                humidity = double.NaN;
+                return false;
+            }
+
+            temperature = st;
+            humidity = srh;
+            return true;
+        }
+
+        private static Register GetMeasurementCmd(bool lowPower, bool clockStretching)
+        {
+            Register mea = Register.SHTC3_MEAS_T_RH_POLLING_NPM;
```

Comment:
rather than allocating this Register, just return for each case the result.
like:
```csharp
if (clockStretching)
{
    return Register.SHTC3_MEAS_T_RH_CLOCKSTR_LPM;
}
```